### ISSUE TITLE
Broken order of setting and checking environment variables

### ISSUE THREAD

**Issue by goldim** [qooxdoo/qooxdoo-compiler#584]
**Compiler version** (qx compiler --version): 1.0.0-beta.20191015-0937

**Tl;dr**: creating and initializing of environment variables happens after test of their existence.

**How did I get it:**  I develop a web client based on OPC UA protocol and use qooxdoo for this goal. For that i took already existed "node-opcua"a library written on typescript, implemented web sockets, did some extra changes and created a bundle "node-opcua" with "browserify" and "babelify". Then I put the resulting script aka bundle to qooxdoo project. I tested this bundle without qooxdoo for browser and like separate nodejs application. There is no any error. Then I debugged the code of qooxdoo application and in bind function of external library node-opcua I stepped into qooxdoo implementation of function bind but in separate application without qooxdoo this was native code. The "qooxdoo " bind is a static method of class _qx.lang.normalize.Function_ which has "defer". In this construction I see the next code:

`if (!qx.core.Environment.get("ecmascript.function.bind"))
 {
Function.prototype.bind = statics.bind;
}
`. 
In overall the method "get" of Environment class checks existence of "_ecmascript.function.bind_" as key in internal map and doesn't find it. If there is this key then the method calls function which really check if "bind" exists and is function. After this I started to look for place in code where the map fills up (initializing) by variables. And put traces there. What I've got the next order:

1. defer of qx.lang.normalize.Function: test of the variable "_ecmascript.function.bind_".
2. defer of qx.bom.client.EcmaScript: qx.core.Environment.add("ecmascript.function.bind", statics.getFunctionBind)

**What I've done to fix this problem for me and my task temporarily**: In generated file part-boot-bundle-1.js commented the line `if (!qx.core.Environment.get("ecmascript.function.bind"))` in defer of _normilize.Function_.

**Comment by hkollmann** [comment]
Could you provide a small example? I tried to reproduce with my own projects - there is defer of qx.bom.client.EcmaScript running first.

**Comment by goldim** [comment]
@hkollmann what sort of examples should I provide? 
I added traces in 2 "defer"s. And with debuger i stepped into the qooxdoo "bind" implementation in the next example of code.

qx.Class.define("qxdemo.Application",
{
  extend : qx.application.Standalone,
  members :
  {
    main : function()
    {      
      this.base(arguments);
      if (qx.core.Environment.get("qx.debug"))
      {        
        qx.log.appender.Native;        
        qx.log.appender.Console;
      }
	    this.g(this.f.bind(this));
    },

    g: function(f){
	    f();
    },

    f: function(){
      console.info("bind");      
    }
  }
});

Maybe a cause of the problem is different software which I use.
I do the next: 
```
$ mkdir qxdemo 
$ cd qxdemo/ 
$ npm i @qooxdoo/compiler @qooxdoo/framework
$ npx qx create qxdemo --out . --type desktop --noninteractive
$ npx qx serve
```
when i installed packages i got: 

> npm WARN bind_test@1.0.0 No description
> npm WARN bind_test@1.0.0 No repository field.
> npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.8 (node_modules/qxcompiler/node_modules/fsevents):
> npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.8: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
> 

My system is :
`qxdemo uname -a 
Linux goldim 4.19.0-5-amd64 #1 SMP Debian 4.19.37-3 (2019-05-15) x86_64 GNU/Linux
`
`cat /etc/debian_version  
10.0`
`qxdemo chromium --version
Chromium 73.0.3683.75 built on Debian buster/sid, running on Debian 10.0`
